groundwater/views.py

```diff
@@ -1 +1,114 @@
-from django.shortcuts import render
+from django.db import transaction
+from django.contrib.gis.geos import Point
+from django.urls import reverse
+from django.views.generic import FormView
+from django.utils import dateparse
+from pyexcel_xls import get_data as xls_get
+from pyexcel_xlsx import get_data as xlsx_get
+from .forms import CsvWellForm
+from .models import GW_Well, GW_GeologyLog
+
+
+class CsvUploadView(FormView):
+    """Upload csv well view.
+    """
+
+    context_object_name = 'csvupload'
+    form_class = CsvWellForm
+    template_name = 'upload_well_csv.html'
+
+    def get_success_url(self):
+        """Define the redirect URL.
+
+       :returns: URL
+       :rtype: HttpResponse
+       """
+
+        return reverse('home_igrac')
+
+    def get_context_data(self, **kwargs):
+        """Get the context data which is passed to a template.
+
+        :param kwargs: Any arguments to pass to the superclass.
+        :type kwargs: dict
+
+        :returns: Context data which will be passed to the template.
+        :rtype: dict
+        """
+
+        context = super(
+            CsvUploadView, self).get_context_data(**kwargs)
+        return context
+
+    def get_form_kwargs(self):
+        """Get keyword arguments from form.
+
+        :returns keyword argument from the form
+        :rtype: dict
+        """
+
+        kwargs = super(CsvUploadView, self).get_form_kwargs()
+        return kwargs
+
+    @transaction.atomic()
+    def post(self, request, *args, **kwargs):
+        """Get form instance from upload.
+
+          :returns: URL
+          :rtype: HttpResponse
+        """
+        form_class = self.get_form_class()
+        form = self.get_form(form_class)
+        gw_location_file = request.FILES.get('gw_location_file')
+        gw_level_file = request.FILES.get('gw_level_file')
+
+        if form.is_valid():
+            if gw_location_file:
+                gw_location_file.seek(0)
+                if str(gw_location_file).split('.')[-1] == "xls":
+                    sheet = xls_get(gw_location_file, column_limit=4)
+                elif str(gw_location_file).split('.')[-1] == "xlsx":
+                    sheet = xlsx_get(gw_location_file, column_limit=4)
+                sheetname = next(iter(sheet))
+                records = sheet[sheetname]
+                for record in records:
```

Comment:
Just remember to wrap this in a single transaction (just as well for both wells an measures)

<img width="672" alt="image" src="https://user-images.githubusercontent.com/178003/85458987-9acd7000-b599-11ea-9927-5f695092594b.png">

Then reject the whole file if anything goes wrong.

Reply:
put into https://github.com/kartoza/IGRAC-GGIS/issues/66